test/Common.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -248,8 +248,38 @@ private void AddEnvironmentVariables(string contentRoot, ApplicationPool pool)
             }
             catch (COMException comException)
             {
-                _logger.LogInformation($"Could not add environment variables to worker process: {comException.Message}");
+                // If we cannot set environment variables on the app pool, set them on the web.config
```

Comment:
I think we shouldn't mix them and automatically fall back. Global environment variables have different effect on some features and tests should be explicit in what they are trying to set.